### PR TITLE
models: include LFS when calculating repo size

### LINKED ISSUE #9369: Repository size in repository management does not account for git LFS data
- Gitea version (or commit ref): 1.10.1
- Git version: 2.17.1
- Operating system: Ubuntu 18.04.3 LTS
- Database (use `[x]`):
  - [x] MySQL
- Can you reproduce the bug at https://try.gitea.io: the IP is unfortunately blocked from Russia.

## Description

The repository size indicated in Repository Management does not account for data stored in git LFS.

So for example, if I create a new repository, add git LFS tracking for zip and add a 20Mb zip file, the repository size shown in the Repository Management is less than 100kb instead of the 20Mb.

It would be better if the size in Repository Management would show the sum of the sizes of git LFS data and repository.

### PR DESCRIPTION
**Pull request by cuonglm**
Fixes #9369

### CODE REVIEW

**Comment by lafriks** [comment]
LFS size should probably be saved in other field imho, also it should include migration to add that to existing repos

**Review by lunny — APPROVED** [review]

**Comment by cuonglm** [comment]
> LFS size should probably be saved in other field imho, also it should include migration to add that to existing repos

Is there any use case for that?

In my case, when I want to limit user for specific size, then I expect repo.Size will return the whole consumed storage of repo.

**Review by 6543 — APPROVED** [review]
split lfs size in a seperate var has it's up and downs.

so as long as we do not explicit need to split this up (because of a feature e.g. ...) we can leafe it by adding lfs to normal repo size

@lafriks do you have a feature/function already in mind?

**Comment by cuonglm** [comment]
The test failed looks weird.

**Comment by 6543** [comment]
ping lgtm

**Comment by cuonglm** [comment]
@6543 your approval is black, is it problem?

**Comment by lafriks** [comment]
We should split them so that we could allow different limits for git repository size and lfs

**Comment by cuonglm** [comment]
> We should split them so that we could allow different limits for git repository size and lfs

Currently, LFS is transparent with user, so I don't think it's worth to split them yet. We can easily do it in the future, if necessary.

**Comment by 6543** [comment]
> @6543 your approval is black, is it problem?

no it just means I cant merge

**Review by sapk — APPROVED** [review]
We should may be add tests for repo.Size but those are not currently existing. Otherwise LGTM.

**Comment by sapk** [comment]
This PR will close on item of #7833. For including or not lfs object in .Size or in a specific field, I think it should be included (as .Size is the global size) and if needed two specific fields could be added for git and lfs specific size info.

**Comment by guillep2k** [comment]
I also like the idea of having separate values for the size. LFS objects might be shared among repositories and reside in different places (e.g. cloud storage), which in turn can result in separate policies, so it makes sense to count them separately (and we can display them in the UI as well).

**Comment by cuonglm** [comment]
> We should may be add tests for repo.Size but those are not currently existing. Otherwise LGTM.

I did add a test in `doLFS`. Can this come in?

**Comment by sapk** [comment]
@cuonglm Yes that is good enough. It is a more broad problem (and previously existing). Currently the only solution to force the updateSize and the size fields to be calculated is to update object in the repo.

**Comment by cuonglm** [comment]
@sapk @lunny Can this come in?

**Comment by lunny** [comment]
@cuonglm I think we could split the size in future PRs.

**Comment by cuonglm** [comment]
@sapk @lunny How is this change going?

There're too many changes in recent master, which makes `doLfs` function gone.

**Comment by techknowlogick** [comment]
ping LG-TM

**Comment by peterbabic** [comment]
Does this change affect https://github.com/go-gitea/gitea/issues/3658 ?

File: models/repo.go [hunk 1]
```diff
@@ -806,6 +806,14 @@ func (repo *Repository) updateSize(e Engine) error {
 		return fmt.Errorf("updateSize: %v", err)
 	}
 
+	objs, err := repo.GetLFSMetaObjects(-1, 0)
```
Comment (lunny) [inline models/repo.go:809]: A select sum sql is better than this change.

**Pull request merged**